indico/modules/events/contributions/lists.py

```diff
@@ -80,28 +82,29 @@ def _filter_list_entries(self, query, filters):
             if status_criteria:
                 criteria.append(db.or_(*status_criteria))
 
-        if 'speakers' in filters['items']:
-            registration_join_criteria = db.and_(
+        if 'people' in filters['items'] or 'speakers' in filters['items']:
+            registration_join_criteria = [
                 Registration.event_id == Contribution.event_id,
                 Registration.is_active,
-                ContributionPersonLink.is_speaker,
                 (Registration.user_id == EventPerson.user_id) | (Registration.email == EventPerson.email)
-            )
+            ]
+            if 'speakers' in filters['items']:
+                registration_join_criteria.append(ContributionPersonLink.is_speaker)
             contrib_query = (db.session.query(Contribution.id)
                              .filter(Contribution.event_id == self.event.id)
                              .join(ContributionPersonLink)
                              .join(EventPerson)
-                             .join(Registration, registration_join_criteria))
+                             .join(Registration, db.and_(*registration_join_criteria)))
             registered_contribs = {id_ for id_, in contrib_query}
 
-            filtered_speakers = filters['items']['speakers']
-            speakers_criteria = []
-            if 'registered' in filtered_speakers:
-                speakers_criteria.append(Contribution.id.in_(registered_contribs))
-            if 'not_registered' in filtered_speakers:
-                speakers_criteria.append(~Contribution.id.in_(registered_contribs))
-            if speakers_criteria:
-                criteria.append(db.or_(*speakers_criteria))
+            people_criteria = []
+            filtered_people = filters['items'].get('people') or filters['items'].get('speakers')
```

Comment:
I don't think this is correct.

For example, take "Speakers: Registered" and "People: Not registered". This will end up showing contributions with unregistered speakers.

You either need to split the code here to be separate for the two filters (ie duplicate the `registered_contribs` logic with the different join criteria), or make the speakers and people filters mutually exclusive. Splitting is probably cleaner since we currently do not have any filters that are exclusive with each other either.

Reply:
For testing this I recommend using an event with contributions like this that cover the combinations:

![image](https://user-images.githubusercontent.com/179599/145810150-92b0ee21-237a-4445-9805-d2ce5981989e.png)

Reply:
While showing unregistered speakers is indeed not correct, I'm not sure it will ever make sense to use these two filters together. In the case where we select "People: Not registered" it will mean selecting any contribution where no one is registered:

![imagem](https://user-images.githubusercontent.com/12183954/148397453-3f164c83-443e-477a-a209-2556d6a7142e.png)

This might seem ambiguous at first, some people might even assume it means selecting any contribution where there is at least someone unregistered:

![imagem](https://user-images.githubusercontent.com/12183954/148397552-dc59febe-3735-4753-b416-11a5412e54bc.png)

In the example above, using "Speakers: Registered" with "People: Not registered" would certainly mean no results as if we are selecting only contributions where no one is registered, we cannot expect a speaker to be registered.

Reply:
~~I agree using them together may not make the most sense.~~

~~However, "Speakers: Registered" and "People: Not registered" would give you contributions where some speakers have registered and some people have not registered...~~

You are right, using both filters at the asme time usually makes no sense, because:

- "Registered" means *someone* from the list of persons is registered
- "Not registered" *nobody* from the list of persons is registered

FWIW I first thought "not registered" means that *someone* from the list is not registered... honestly I'm not sure what behavior makes more sense, but "nobody" registered is probably more useful for organizers since what they typically care about is that there's someone to give the presentation / explain the poster...